scheduler.go

```diff
@@ -30,9 +30,10 @@ type Scheduler struct {
 
 	tags sync.Map // for storing tags when unique tags is set
 
-	tagsUnique      bool // defines whether tags should be unique
-	updateJob       bool // so the scheduler knows to create a new job or update the current
-	waitForInterval bool // defaults jobs to waiting for first interval to start
+	tagsUnique                     bool // defines whether tags should be unique
+	updateJob                      bool // so the scheduler knows to create a new job or update the current
+	waitForInterval                bool // defaults jobs to waiting for first interval to start
+	isSetBehaviourBeforeJobCreated bool // represents if set some feature before job created
```

Comment:
```suggestion
	tagsUnique      bool // defines whether tags should be unique
	updateJob       bool // so the scheduler knows to create a new job or update the current
	waitForInterval bool // defaults jobs to waiting for first interval to start
	jobCreated      bool // so the scheduler knows a job was created prior to calling Every or Cron
```

Reply:
@denizzengin if you can make this change and refactor all the references, then I'll approve!

Reply:
@JohnRoesler  it's done but there is a test error but i run local tests successfully - (with make test)
Actually, my changes don't releated with TestScheduler_SetMaxConcurrentJobs method but gave error. i don't know what's going on here. Any suggestion ?

Reply:
occaisionally tests fail due to timing / slowness in the docker containers that actions uses. I've been meaning to open an issue for us to refactor the tests to use a minimum of 1 second duration because I have some that use milliseconds interval in testing and it's not reliable in the actions pipeline